cpp/include/cudf/lists/drop_list_duplicates.hpp

```diff
@@ -28,35 +28,102 @@ namespace lists {
  */
 
 /**
- * @brief Create a new lists column by extracting unique entries from list elements in the given
- * lists column.
+ * @brief Create new lists columns by copying entries from the input lists columns ignoring
+ * duplicate list entries.
  *
- * Given an input lists column, the list elements in the column are copied to an output lists
- * column such that their duplicated entries are dropped out to keep only the unique ones. The
- * order of those entries within each list are not guaranteed to be preserved as in the input. In
- * the current implementation, entries in the output lists are sorted by ascending order (nulls
- * last), but this is not guaranteed in future implementation.
+ * Given a pair of keys-values lists columns. At each row, each list entry in one column maps
+ * to a list entry in the other column by linear order and vice versa thus lists at each row
+ * in both keys and values columns always have the same size. A pair of output keys and values lists
+ * columns are generated by copying entries from the input lists columns. The duplicate entries in
+ * each list of the keys column are ignored from copying to output only the unique list entries.
+ * Entries in the values column mapped to these key entries are also copied or ignored along with
+ * them.
  *
- * @throw cudf::logic_error if the child column of the input lists column contains nested type other
- * than struct.
+ * The users are responsible to have the keys-values lists columns having the same number of entries
+ * at each row. Otherwise, crash may happen, or the results will be undefined.
```

Comment:
```suggestion
 * The users are responsible for the keys-values lists columns having the same number of entries
 * at each row. Otherwise, a crash may happen, or the results will be undefined.
```